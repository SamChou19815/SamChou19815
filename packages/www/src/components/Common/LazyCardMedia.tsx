import type { ReactElement } from 'react';

type Props = {
  readonly image: string;
  readonly title: string;
};

const LazyCardMedia = ({ image, title }: Props): ReactElement => (
  <div className="card__image">
    <img src={image} alt={title} title={title} loading="lazy" />
  </div>
);

export default LazyCardMedia;
