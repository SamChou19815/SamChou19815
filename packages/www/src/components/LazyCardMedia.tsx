import React from 'react';

type Props = {
  readonly image: string;
  readonly title: string;
};

const LazyCardMedia = ({ image, title }: Props): JSX.Element => (
  <div className="card__image">
    <img src={image} alt={title} title={title} loading="lazy" />
  </div>
);

export default LazyCardMedia;
