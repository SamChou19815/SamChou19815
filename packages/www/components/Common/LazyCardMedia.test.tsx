import React from 'react';

import renderer from 'react-test-renderer';

import LazyCardMedia from './LazyCardMedia';

it('LazyCardMedia matches snapshot.', () => {
  const tree = renderer.create(<LazyCardMedia image="test" title="title" />).toJSON();
  expect(tree).toMatchSnapshot();
});
