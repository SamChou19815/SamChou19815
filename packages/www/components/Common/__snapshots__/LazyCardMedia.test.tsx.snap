// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyCardMedia matches snapshot. 1`] = `
<div
  className="card__image"
>
  <img
    alt="title"
    loading="lazy"
    src="test"
    title="title"
  />
</div>
`;
